### PR TITLE
Don't hit sign-in endpoints for api-key protected endpoints

### PR DESCRIPTION
**Pull request by timabell**
### Context

API was hitting sign-in end point for every api-key based request causing extra load, errors and confusion.

### Changes proposed in this pull request

* Remove default schemes
* Specify scheme on every endpoint

### CODE REVIEW

File: src/ManageCourses.Api/Controllers/AcceptTermsController.cs [hunk 1]
```diff
@@ -17,7 +18,7 @@ public AcceptTermsController(IManageCoursesDbContext context)
             this.context = context;
         }
 
-        [Authorize]
+        [Authorize(AuthenticationSchemes = BearerTokenDefaults.AuthenticationScheme)]
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/AcceptTermsController.cs]: Can we create a class that inherits from `Authorize` ie `BearerTokenAuthorize` and its counterpart ``BearerTokenApiKeyAuthorize``

So it shows up as a hard choice made therefore usage of `Authorize` can be quickly discovered quickly

File: src/ManageCourses.Api/Startup.cs [hunk 2]
```diff
@@ -52,17 +52,15 @@ public void ConfigureServices(IServiceCollection services)
...
-                options.DefaultAuthenticateScheme = BearerTokenDefaults.AuthenticationScheme;
-                options.DefaultChallengeScheme = BearerTokenDefaults.AuthenticationScheme;
-            }).AddBearerToken(options =>
-            {
-                options.UserinfoEndpoint = Configuration["auth:oidc:userinfo_endpoint"];
-            }).AddBearerTokenApiKey(options =>
-            {
-                options.ApiKey = Configuration["api:key"];
-            });
+            services.AddAuthentication()
```
Comment (defong) [inline src/ManageCourses.Api/Startup.cs]:
```
services.AddAuthentication(
 // And explain it here of the available choices
 // As AuthorizeAttribute was designed to use a default that we no longer set.
)
```

File: src/ManageCourses.Api/Controllers/AcceptTermsController.cs (hunk 1 above)
Comment (timabell) [reply src/ManageCourses.Api/Controllers/AcceptTermsController.cs]: done. good call :+1:

File: src/ManageCourses.Api/Startup.cs (hunk 2 above)
Comment (timabell) [reply src/ManageCourses.Api/Startup.cs]: Not sure what to write here, suggestions?

Comment (defong) [reply src/ManageCourses.Api/Startup.cs]: use one of the following?

Comment (defong) [reply src/ManageCourses.Api/Startup.cs]: BearerTokenAuthorize and its counterpart BearerTokenApiKeyAuthorize as `Authorize` is dead?

Comment (timabell) [reply src/ManageCourses.Api/Startup.cs]: done

**Pull request merged**